extensions-core/druid-bloom-filter/src/main/java/org/apache/druid/query/aggregation/bloom/BloomFilterAggregatorFactory.java

```diff
@@ -0,0 +1,233 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.bloom;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.druid.io.ByteBufferInputStream;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.query.ColumnSelectorPlus;
+import org.apache.druid.query.aggregation.Aggregator;
+import org.apache.druid.query.aggregation.AggregatorFactory;
+import org.apache.druid.query.aggregation.AggregatorUtil;
+import org.apache.druid.query.aggregation.BufferAggregator;
+import org.apache.druid.query.aggregation.bloom.types.BloomFilterAggregatorColumnSelectorStrategy;
+import org.apache.druid.query.aggregation.bloom.types.BloomFilterAggregatorColumnSelectorStrategyFactory;
+import org.apache.druid.query.cache.CacheKeyBuilder;
+import org.apache.druid.query.dimension.DimensionSpec;
+import org.apache.druid.segment.ColumnSelectorFactory;
+import org.apache.druid.segment.DimensionHandlerUtils;
+import org.apache.hive.common.util.BloomKFilter;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+public class BloomFilterAggregatorFactory extends AggregatorFactory
+{
+  protected static final BloomFilterAggregatorColumnSelectorStrategyFactory STRATEGY_FACTORY =
+      new BloomFilterAggregatorColumnSelectorStrategyFactory();
+
+  private final String name;
+  private final DimensionSpec field;
+  private final int maxNumEntries;
+
+  @JsonCreator
+  public BloomFilterAggregatorFactory(
+      @JsonProperty("name") String name,
+      @JsonProperty("field") final DimensionSpec field,
+      @JsonProperty("maxNumEntries") Integer maxNumEntries
+  )
+  {
+    this.name = name;
+    this.field = field;
+    this.maxNumEntries = maxNumEntries != null ? maxNumEntries : 1500;
+  }
+
+  @Override
+  public Aggregator factorize(ColumnSelectorFactory columnFactory)
+  {
+    ColumnSelectorPlus<BloomFilterAggregatorColumnSelectorStrategy> selectorPlus =
+        DimensionHandlerUtils.createColumnSelectorPlus(
+            STRATEGY_FACTORY,
+            field,
+            columnFactory
+        );
+
+    return new BloomFilterAggregator(selectorPlus, maxNumEntries);
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(ColumnSelectorFactory columnFactory)
+  {
+    ColumnSelectorPlus<BloomFilterAggregatorColumnSelectorStrategy> selectorPlus =
+        DimensionHandlerUtils.createColumnSelectorPlus(
+            STRATEGY_FACTORY,
+            field,
+            columnFactory
+        );
+
+    return new BloomFilterBufferAggregator(selectorPlus, maxNumEntries);
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    // idk how to compare?
+    return (Comparator<BloomKFilter>) (o1, o2) -> 0;
```

Comment:
Could compare the number of set bits.

Reply:
:+1:

Reply:
Added method to copied `BloomKFilter` to count the number of set bits, allowing ordering results by density of bloom filter